src/dos/dos_mscdex.cpp

```diff
@@ -1093,9 +1103,30 @@ static Bit16u MSCDEX_IOCTL_Optput(PhysPt buffer,Bit8u drive_unit) {
 
 static MountType MSCDEX_GetMountType(const char *path)
 {
+	const char *cdName;
+	char buffer[512];
+	strcpy(buffer, path);
+#if defined(WIN32)
+	upcase(buffer);
+#endif
```

Comment:
Same as before regarding using `std::string`, and the necessity of `upcase()` on Windows only.

Reply:
Done.

Reply:
The unbounded `strcpy` was also flagged by Coverity:

![2022-02-14_14-46](https://user-images.githubusercontent.com/1557255/153959466-3c0b3577-3622-4d62-8123-74b50e407e70.png)

Suggest going back to @shermp's recommendation.